### ISSUE TITLE
Add forward compatibility to TaggedFieldSerializer

### ISSUE THREAD

**Issue by causis** [EsotericSoftware/kryo#352]
Hello, how do you do?
I was wondering why TaggedFieldSerializer doesn't have a configuration like jackson's "FAIL_ON_UNKNOWN_PROPERTIES". It seems that it would be an easy way to add forward compatibility on demand, and I think the only piece of code that would need to be changed is:
```
if (cachedField == null) 
  throw new KryoException("Unknown field tag: " + tag + " (" + getType().getName() + ")");
```
This would give us more flexibility, specially when 2 different apps are used to serialize and deserialize the object.
It would also allow to avoid having to declare deprecated fields, although this could be a dangerous practice, leading to forgetting and reusing a tag.
Thank you very much!

**Comment by psmarcos** [comment]
+1

**Comment by magro** [comment]
Pull requests welcome! :-)

Am 01.10.2015 21:41 schrieb "Pablo Marcos" <notifications@github.com>:

**Comment by causis** [comment]
Great! I'll work on it and send it!
Thanks!

**Comment by levyfan** [comment]
Hi, I have just sent a pull request. Could you please help review it?

**Comment by fgaule** [comment]
Would be great to have this feature soon :+1:

**Comment by magro** [comment]
Resolved with #365

**Comment by cypherdare** [comment]
@levyfan Could you explain why the fields need to be written in tag value order? And I don't see where or how the unknown bytes of the future tagged field are skipped in the Input. Thanks.
